### ISSUE TITLE
[0.7.0] yo polymer returns error

### ISSUE THREAD

**Issue by moderndeveloperllc** [yeoman/generator-polymer#147]
```
$ sudo yo polymer
/opt/local/lib/node_modules/generator-polymer/node_modules/yeoman-generator/lib/base.js:444
  this.env.sharedFs.on('change', writeFiles);
                    ^
TypeError: undefined is not a function
    at run (/opt/local/lib/node_modules/generator-polymer/node_modules/yeoman-generator/lib/base.js:444:21)
    at Environment.run (/opt/local/lib/node_modules/yo/node_modules/yeoman-environment/lib/environment.js:330:20)
    at /opt/local/lib/node_modules/yo/lib/cli.js:91:9
    at Environment.resolver.lookup (/opt/local/lib/node_modules/yo/node_modules/yeoman-environment/lib/resolver.js:50:12)
    at init (/opt/local/lib/node_modules/yo/lib/cli.js:66:7)
    at pre (/opt/local/lib/node_modules/yo/lib/cli.js:53:3)
    at Object.<anonymous> (/opt/local/lib/node_modules/yo/lib/cli.js:155:1)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
```

On OSX 10.10.2. The update to 0.7 appeared to work well:
```
sudo npm install -g generator-polymer
generator-polymer@0.7.0 /opt/local/lib/node_modules/generator-polymer
├── rimraf@2.2.8
├── ncp@1.0.1
├── chalk@0.5.1 (escape-string-regexp@1.0.2, ansi-styles@1.1.0, supports-color@0.2.0, strip-ansi@0.3.0, has-ansi@0.1.0)
├── yosay@1.0.2 (ansi-regex@1.1.0, string-length@1.0.0, ansi-styles@2.0.0, strip-ansi@2.0.1, pad-component@0.0.1, word-wrap@1.0.0, minimist@1.1.0, taketalk@1.0.0)
├── validate-element-name@1.0.0 (log-symbols@1.0.1, ncname@1.0.0)
├── lodash@2.4.1
└── yeoman-generator@0.18.9 (yeoman-welcome@1.0.0, read-chunk@1.0.1, dargs@3.0.1, detect-conflict@1.0.0, xdg-basedir@1.0.1, yeoman-assert@1.0.0, user-home@1.1.1, diff@1.2.2, class-extend@0.1.1, pretty-bytes@1.0.3, text-table@0.2.0, mime@1.3.4, underscore.string@2.4.0, async@0.9.0, istextorbinary@1.0.2, debug@2.1.1, nopt@3.0.1, cross-spawn@0.2.6, run-async@0.1.0, mem-fs-editor@1.2.1, mkdirp@0.5.0, shelljs@0.3.0, through2@0.6.3, cli-table@0.3.1, dateformat@1.0.11, github-username@1.1.1, glob@4.4.0, findup-sync@0.2.1, file-utils@0.2.2, download@3.3.0, gruntfile-editor@0.2.0, sinon@1.12.2, cheerio@0.18.0, yeoman-environment@1.2.4, inquirer@0.8.0)
```

**Comment by robdodson** [comment]
You shouldn't need to run `sudo` with `yo polymer`. What happens if you try without that?

**Comment by moderndeveloperllc** [comment]
Same. I used sudo after it failed the first time. Any logging I can turn on to see what I'm missing?

**Comment by robdodson** [comment]
What happens if you update `yo`?

`npm install -g yo`

Just guessing based on this ticket: https://github.com/yeoman/generator-polymer/issues/145

**Comment by moderndeveloperllc** [comment]
Interesting. Looks like `yo` was not installed? Pretty sure I had sone so, but it might have been a local install rather than global. Sorry for the troubles. BTW, I'm using polymer 0.5.5 for a different, internal project, and I'm quite liking it! Y'all keep up the great work.

**Comment by alexpaluzzi** [comment]
I'm having the same issue now and I've used the generator 20 times. I'm not sure what has changed. (Yeoman is definitely installed and everything is updated).

**Comment by robdodson** [comment]
hm, @addyosmani @sindresorhus @SBoudrias any ideas? The latest release [upgraded to `yeoman-generator` 0.18+](https://github.com/yeoman/generator-polymer/blob/master/package.json#L29), I wonder if that has anything to do with it...?

**Comment by SBoudrias** [comment]
Oh yeah, this mean you need to update yo. `npm -g update yo`

**Comment by alexpaluzzi** [comment]
@robdodson @SBoudrias so, it is definitely because of an out-of-date Yeoman, as you guys have mentioned.

Before commenting here originally, I made sure to update everything. `--version` was printing the latest of everything. Digging further, I found multiple folders with the Yo binary in it, and I think this happened as a result of a recent `npm` update. I was updating Yo but when I was calling it, it was calling an older version that was further up the chain in the PATH. I deleted the old stuff and it worked fine.

This seems to be a really common issue with recent Node and NPM updates. So it is not related to this generator. Wanted to mention it to you guys in-case someone else comes along with the same problem (even after "updating").

Thanks
